server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -1364,5 +1368,47 @@ public void calculateExportedForOwnersPools(Owner owner) {
         q.executeUpdate();
     }
 
+    public void markCertificatesDirtyForPoolsWithProducts(Owner owner, Collection<String> productIds) {
+        if (productIds.size() <= MAX_IN_CLAUSE_VALUES) {
+            markCertificatesDirtyForPoolsWithNormalProducts(owner, productIds);
+            markCertificatesDirtyForPoolsWithProvidedProducts(owner, productIds);
+        }
+        else {
+            Iterator<String> productIdIter = productIds.iterator();
+            List<String> batch = new ArrayList<String>();
```

Comment:
We should use the Iterables.partition method here, as it creates a view over sections of the original collection instead of having to copy over blocks to a new list. We can reduce this down to (something close to) the following example:

```
for (List<String> block : Iterables.partition(productIds, IN_OPERATOR_BLOCK_SIZE)) {
    this.markCertificatesDirtyForPoolsWithNormalProducts(owner, block);
    this.markCertificatesDirtyForPoolsWithProvidedProducts(owner, block);
}
```